mne/time_frequency/spectrum.py

```diff
@@ -1179,6 +1179,56 @@ def __getitem__(self, item):
         return BaseRaw._getitem(self, item, return_times=False)
 
 
+@fill_doc
+class SpectrumArray(Spectrum):
+    """Spectrum object from numpy array.
+
+    Parameters
+    ----------
+    data : array, shape (n_channels, n_freqs)
+        The channels' power spectral density for each epoch.
+    %(info_not_none)s
+    %(freqs_tfr)s
```

Comment:
I took out method but `mne.time_frequency.psd_array_welch/multitaper` return `freqs` and maybe it's a bit paranoid but adding a round trip from `freqs` to `fmin`/`fmax` and back to `freqs` one more place for sampling rate or what have you to get mixed up. Also, I cut out power noise so the `freqs` are actually modified and that wouldn't work for my use case. If that's not convincing, I'm fine doing `fmin`, `fmax` and then modifying the `psd` object for my use case if it's important to you that the API matches.

Reply:
> I cut out power noise so the `freqs` are actually modified and that wouldn't work for my use case

So you mean you have bins that are missing / your bins are not equally spaced?  I guess that's a compelling use case.  And since folks aren't really supposed to use the `(Epochs)Spectrum` constructors directly, I'm OK to sacrifice API similarity in this case.